### PR TITLE
If TDSCatalog detects html, warn and try an XML link.

### LINKED ISSUE #29: Handle errors when passing HTML to `TDSCatalog`
Right now, if accidentally pass an html catalog to `TDSCatalog`, we get:
```
  File "<string>", line unknown
ParseError: mismatched tag: line 7, column 66
```
I don't know if it's a good idea to rewrite the URL, but we should at least detect it somehow (maybe using the content type on the headers) and throw a clear error.

### PR DESCRIPTION
**Pull request by dopplershift**
Fixes #29.

### CODE REVIEW

File: siphon/catalog.py [hunk 1]
```diff
@@ -41,11 +41,23 @@ def __init__(self, catalog_url):
...
+
+        session = create_http_session()
+
         # get catalog.xml file
-        xml_fobj = urlopen(catalog_url)
+        resp = session.get(self.catalog_url)
+
+        # If we were given an HTML link, warn about it and try to fix to xml
+        if 'html' in resp.headers['content-type']:
+            import warnings
```
Comment (lesserwhirls) [inline siphon/catalog.py]: Why the warnings package and not logging.warning?

Comment (dopplershift) [inline siphon/catalog.py]: The logic that was suggested before was: logging for warnings of bad conditions that the user may want to know about, but can't do anything about; warnings module for warnings the user should address. 

https://docs.python.org/3/howto/logging.html

Specifically: "warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning"

**Comment by dopplershift** [comment]
My bigger question, is automatically trying a new URL and setting that on the `TDSCatalog` instance a good idea?

File: siphon/catalog.py (hunk 1 above)
Comment (lesserwhirls) [inline siphon/catalog.py]: Ah yes, I remember now.

**Comment by lesserwhirls** [comment]
I think it's ok to try with xml if the url ends with html. It's a very common cut/paste error when trying to find a catalog url.

**Comment by dopplershift** [comment]
Not just url endswith html, I replace all 'html' in the URL with 'xml'. This is important for URLs like: http://thredds.ucar.edu/thredds/catalog/grib/NCEP/GFS/Global_onedeg/catalog.html?dataset=grib/NCEP/GFS/Global_onedeg/Best

**Comment by lesserwhirls** [comment]
Ah. Yes. I think that is still ok to do.

**Pull request merged**